pkg/statistics/histogram.go

```diff
@@ -1049,22 +1073,26 @@ func (hg *Histogram) OutOfRangeRowCount(
 		upperBound = hg.NotNullCount() / float64(histNDV)
 	}
 
-	allowUseModifyCount := sctx.GetSessionVars().GetOptObjective() != variable.OptObjectiveDeterminate
-
 	if !allowUseModifyCount {
 		// In OptObjectiveDeterminate mode, we can't rely on the modify count anymore.
 		// An upper bound is necessary to make the estimation make sense for predicates with bound on only one end, like a > 1.
 		// We use 1/NDV here (only the Histogram part is considered) and it seems reasonable and good enough for now.
 		return min(rowCount, upperBound)
 	}
 
-	// If the modifyCount is large (compared to original table rows), then any out of range estimate is unreliable.
+	// If the realtimeRowCount is larger than the original table rows, then any out of range estimate is unreliable.
 	// Assume at least 1/NDV is returned
-	if float64(modifyCount) > hg.NotNullCount() && rowCount < upperBound {
-		rowCount = upperBound
-	} else if rowCount < upperBound {
-		// Adjust by increaseFactor if our estimate is low
-		rowCount *= increaseFactor
+	addedRows := float64(realtimeRowCount) - hg.TotalRowCount()
+	if addedRows > 1 {
+		// Conservatively - use the larger of the left or right percent - since we are working with
+		// changes to the table since last Analyze - any out of range estimate is unreliable.
+		// if the histogram range is invalid (too small/large - histInvalid) - totalPercent is zero
+		// and we will set rowCount to min of upperbound and added rows
+		totalPercent = min(0.5, max(leftPercent, rightPercent))
```

Comment:
May I ask why 0.5?

Reply:
I guess it's just an arbitrary number to make the final estimated results not too aggressive.

Reply:
I used 0.5 because I did not want to always assume that any new rows added, or updates to the existing rows were all changing the high end 100% or the low end 100%. If you started with 100 rows, and you added 10 rows - you could've added 5 rows outside the low bound, and 5 rows above the high bound. Or all above or below - or all within the range. Assuming all rows added are 100% in one direction cannot be verified - therefore - 0.5 is used.